### PR TITLE
PlayGo: basic implementation, credits to red-prig

### PR DESCRIPTION
**Pull request by bax-cz**
Some more PlayGo implementation inspired by red-prig.

### CODE REVIEW

**Comment by georgemoralis** [comment]
looking good any specific game tested? i think infamous first (or second) needed that . Worms Also needed this but was working with the previous implementation

**Comment by Graubek** [comment]
It fixed  [Lib.PlayGo] <Error> playgo.cpp:scePlayGoGetLocus:53: (STUBBED)called handle = 1, chunkIds = 2, numberOfEntries = 69  loop in InFamous second son and first light

**Comment by bax-cz** [comment]
Yep, as 

> It fixed [Lib.PlayGo] playgo.cpp:scePlayGoGetLocus:53: (STUBBED)called handle = 1, chunkIds = 2, numberOfEntries = 69 loop in InFamous second son and first light

Yes, this. But still won't boot :(

File: src/core/file_format/playgo_chunk.h [hunk 1]
```diff
@@ -3,29 +3,128 @@
 
 #pragma once
 #include <filesystem>
-#include "common/types.h"
+#include <mutex>
```
Comment (georgemoralis) [inline src/core/file_format/playgo_chunk.h:6]: probably needs an #include <vector>  to be able to compile under linux as well

Comment (bax-cz) [reply src/core/file_format/playgo_chunk.h:6]: ok, I'm in the process of setting up a linux build environment, this might take a while..

Comment (georgemoralis) [reply src/core/file_format/playgo_chunk.h:6]: No needed just add an #include <vector> and it should work

Comment (georgemoralis) [reply src/core/file_format/playgo_chunk.h:6]: vector

Comment (bax-cz) [reply src/core/file_format/playgo_chunk.h:6]: ok, should be fixed now

**Comment by george24f** [comment]
I m a bit stuck on this what can i do to fix my infamous second son

**Pull request merged**